test/modules/WebSocket/compliance/websocket.connection.test.ts

```diff
@@ -15,30 +16,35 @@ afterAll(() => {
   interceptor.dispose()
 })
 
-it('emits the correct connection event', async () => {
+it('emits the correct "connection" event on the interceptor', async () => {
   const connectionListener = vi.fn()
   interceptor.once('connection', connectionListener)
 
   new WebSocket('wss://example.com')
 
-  await vi.waitFor(() => {
-    expect(connectionListener).toHaveBeenCalledTimes(1)
-    expect(connectionListener).toHaveBeenNthCalledWith(
-      1,
-      expect.objectContaining({
-        client: expect.objectContaining({
-          id: expect.stringMatching(UUID_REGEXP),
-          send: expect.any(Function),
-          addEventListener: expect.any(Function),
-          removeEventListener: expect.any(Function),
-          close: expect.any(Function),
-        }),
-        server: expect.objectContaining({
-          send: expect.any(Function),
-          addEventListener: expect.any(Function),
-          removeEventListener: expect.any(Function),
-        }),
-      })
-    )
-  })
+  // Must not emit the "connection" event on this tick.
+  expect(connectionListener).toHaveBeenCalledTimes(0)
+
+  await waitForNextTick()
```

Comment:
Also modifying this compliance test to ensure the connection event is emitted on the next tick, not the current one.